librz/include/rz_endian.h

```diff
@@ -646,24 +1581,192 @@ static inline void rz_write_ble(void *dst, ut64 val, bool big_endian, int size)
+ */
+static inline void rz_write_at_ble64(void *dest, ut64 val, bool big_endian, size_t offset) {
+	if (big_endian) {
+		rz_write_at_be64(dest, val, offset);
+	} else {
+		rz_write_at_le64(dest, val, offset);
+	}
+}
+
+/**
+ * \brief Write a 128-bit value to \p dest in specified order.
+ * \param[out] dest The pointer to which the value is written.
+ * \param val The written 128-bit value.
+ * \param big_endian The choice of endianness.
+ * \param offset The offset at which the value is written.
+ *
+ * The value is written according to \p big_endian. If \p big_endian
+ * is \c true, then an integer in big-endian order is
+ * written. Otherwise, if \p big_endian is \c false, an integer in
+ * little-endian order is written.
+ */
+static inline void rz_write_at_ble128(void *dest, ut128 val, bool big_endian, size_t offset) {
+	if (big_endian) {
+		rz_write_at_be128(dest, val, offset);
+	} else {
+		rz_write_at_le128(dest, val, offset);
+	}
+}
+
+/**
+ * \brief Write a 32-bit floating-point to \p dest in specified order.
+ * \param[out] dest The pointer to which the floating-point value is written.
+ * \param val The floating-point value to be written.
+ * \param big_endian The choice of endianness.
+ * \param offset The offset at which the value is written.
+ *
+ * The value is written according to \p big_endian. If \p big_endian
+ * is \c true, then a floating-point value in big-endian order is
+ * written. Otherwise, if \p big_endian is \c false, a floating-point
+ * value in little-endian order is written.
+ */
+static inline void rz_write_at_ble_float(void *src, float val, bool big_endian) {
+	big_endian ? rz_write_be_float(src, val) : rz_write_le_float(src, val);
+}
+
+/**
+ * \brief Write a 64-bit floating-point to \p dest in specified order.
+ * \param[out] dest The pointer to which the floating-point value is written.
+ * \param val The floating-point value to be written.
+ * \param big_endian The choice of endianness.
+ * \param offset The offset at which the value is written.
+ *
+ * The value is written according to \p big_endian. If \p big_endian
+ * is \c true, then a floating-point value in big-endian order is
+ * written. Otherwise, if \p big_endian is \c false, a floating-point
+ * value in little-endian order is written.
+ */
+static inline void rz_write_at_ble_double(void *src, double val, bool big_endian) {
+	big_endian ? rz_write_be_double(src, val) : rz_write_le_double(src, val);
+}
+
 /*swap*/
 
 /* Use compiler intrinsics if present */
 
+/**
+ * \def rz_swap_ut16
+ * \brief Reverse the order of bytes of the 16-bit representation.
+ * \parameter x The 16-bit value to operate upon.
+ *
+ * # Example
+ *
+ * \code{.c}
+ * ut16 x = 0xAABB;
+ * ut16 result = 0xBBAA;
+ * assert(rz_swap_ut16(x) == result);
+ * \endcode
+ */
 #if HAVE___BUILTIN_BSWAP16
 #define rz_swap_ut16 __builtin_bswap16
 #else
 static inline ut16 rz_swap_ut16(ut16 val) {
 	return (val << 8) | (val >> 8);
 }
+#define rz_swap_ut16 rz_swap_ut16
```

Comment:
What is this for?

Reply:
With Doxygen I document the `rz_swap_ut16` macro. I wasn't sure if the macro needed to be present or not, so I made sure it was always defined. Maybe I should just remove the extra `#define` and keep the Doxygen comment as it is?

Reply:
Yes, just remove the macro